_books/ion-1-1/src/modules/encoding_modules.md

```diff
@@ -0,0 +1,228 @@
+        (macro bar () Bar)))
+
+// `_` now contains macros `foo` and `bar`
+
+(:foo) // Equivalent to `(:_::foo)`
+(:bar) // Equivalent to `(:_::bar)`
+```
+
+System macros like [`add_symbols`](../macros/system_macros.md#add_symbols)
+and [`add_macros`](../macros/system_macros.md#add_macros) apply their changes to `_`,
+so we can rewrite the above more succinctly as:
+```ion
+$ion_1_1
+
+// `_` exists, but is empty
+
+(:add_macros
+    (macro foo () Foo)
+    (macro bar () Bar))
+
+// `_` now contains macros `foo` and `bar`
+
+(:foo) // Equivalent to `(:_::foo)`
+(:bar) // Equivalent to `(:_::bar)`
+```
+
+## Default encoding module sequence
+
+At the beginning of a stream, the encoding module sequence contains two modules:
+1. the [default module](#default-module), `_`
+2. the [system module](system_module.md), `$ion`
+
+Recall that a segment's symbol and macro tables are logical concatenations of those found in the segment's encoding modules.
+Because `_` is empty at the beginning of the stream,
+the stream's initial symbol and macro tables are identical to those of the system module, `$ion`.
+
+This is beneficial because it allows all system macros to be invoked from the stream's macro table
+[in a single byte](../binary/e_expressions.md#e-expression-with-the-address-in-the-opcode)
+rather than [the two-byte sequence](http://localhost:3000/binary/e_expressions.html#system-macro-invocations)
+needed to invoke them from the system macro table.
+In this way, a writer can define its macros and symbols in a maximally compact fashion at the head of the stream.
+
+## Modifying active modules
+
+If a module binding in the encoding module sequence is redefined,
+the new module definition replaces the old one in the sequence.
+
+For example after these directives are evaluated:
+```ion
+$ion::
+(module mod_a
+    (macro_table
+        (macro foo () Foo))
+        (macro bar () Bar)))
+
+$ion::
+(module mod_b)
+
+$ion::
+(module mod_c
+    (macro_table
+        (macro quux () Quux)
+        (macro quuz () Quuz)))
+
+$ion::(encoding mod_a mod_b mod_c)
+```
+the encoding sequence is `_ mod_a mod_b mod_c`, and `mod_b` is empty.
+
+```ion
+(.0) // => Foo
+(.1) // => Bar
+(.2) // => Quux
+(.3) // => Quuz
+```
+
+If we then add macros to `mod_b`, those macros will immediately become available.
+
+```ion
+$ion::
+(module mod_b
+    (macro_table
+        (macro baz () Baz)))
+
+(.0) // => Foo
+(.1) // => Bar
+(.2) // => Baz
+(.3) // => Quux
+(.4) // => Quuz
+```
+
+> [!IMPORTANT]
+> Notice that modifying a module (in this case `mod_b`) can cause the addresses of all subsequent macros to be modified.
+
+## Clearing the symbol and macro tables
+```ion
+(module _) // Redefine `_` to be an empty module
+// If other modules are in use, remove them from the encoding module sequence
+$ion::(encoding)
+```
+
+Note that this is different from the behaviour of an IVM.
+When an IVM is encountered, the encoding module is set to the system module.
```

Comment:
This statement is unclear. Do you mean that the default module is populated with the content of the system module?

Edit: on further thought, I think you mean to say that "the encoding module _sequence_ is set to the system module" (or something like that). Although earlier in this document, you say

> At the beginning of a stream, the encoding module sequence contains two modules:
> 1. the [default module](#default-module), `_`
> 2. the [system module](system_module.md), `$ion`

Reply:
This sentence is a holdover from the singleton-encoding-module model of things. I'll fix it, thanks.